### ISSUE TITLE
Show message when trying to toggle the diff view outside of a diff

### ISSUE THREAD

**Issue by lope** [samu/git-diff-details#62]
Hi, I'm running Atom 1.9.8 on x86_64 linux

I've been struggling with this for at least 10 mins. It came up initially a few times but seemed to be conflicting with the native package: 
I've restarted atom each time and also done a atom refresh each time (ctrl alt R)

I've tried the defaults,
alt+g alt+d

Interpreted in every possible way
1. (meaning: hold alt, press g, release both, hold alt press d)
2. (hold alt press g, keep holding alt, press d)
3. (hold alt, press g and d and hold all 3 at the same time)

I've also tried setting it as custom keys
'ctrl-h':  'git-diff-details:toggle-git-diff-details';
'alt-d':  'git-diff-details:toggle-git-diff-details';
'ctrl-alt-d':  'git-diff-details:toggle-git-diff-details';
'ctrl-alt-shift-d':  'git-diff-details:toggle-git-diff-details';
'alt-shift-d':  'git-diff-details:toggle-git-diff-details';

I've got custom keys that I've set for other things, so custom keys do work for other things for me. I've also double checked that I'm not using any of the keys tested as my desktop's global shortcuts.

I've tried disabling the core packages: git-diff open-on-github.

Basically can't use it :/

Any ideas?

**Comment by samu** [comment]
@lope please note you have to place your cursor on an area that actually has diffs. See #60

**Comment by lope** [comment]
@samu thank you very much for your kind and helpful response and sharing this package.

I see that it does work, although it does nothing for the lines that are added (the green plus instead of the yellow circle), which of course is understandable and reasonable now that I'm familiar with it's interface, know what it does and what to expect from it.

One piece of constructive criticism and suggestion I can make is that the UX is a little bit difficult for people who are unfamiliar with this package as to what exactly it does, and the requirement of the cursor being over a modified line. It's extremely frustrating pressing hotkeys and experiencing absolutely nothing happening. It feels like the hotkeys aren't working or the package is not working. To receive absolutely no response by default, in my opinion is a huge UX problem.

I suggest that if there's a nop action (do nothing because the user does not have the cursor on a modified line), that you pop up a msgbox telling them they need to be on a modified line. It will save a lot of time, frustration and github issues/comments.

However personally I'm quite glad it didn't work for me, because it facilitated me trying split-diff which is more in-line with my own preferences/workflow, so personally I will use that now.

Thanks once again for sharing this package and responding in such a helpful and timely manner!

**Comment by samu** [comment]
@lope thanks for the response. I'll actually keep this issue open as i think the idea of displaying a message box when the user is outside of a diff is great.

**Comment by lope** [comment]
@samu thanks sounds good, after a moment's reflection in Atom I can actually see a use case for using _git-diff-details_ and _split-diff_ packages.

_git-diff-details_ might be better when coding with a few vertical splits and there's no space to fit _split-diff_'s extra pane. Then _split-diff_ can provide the detailed view when needed.

**Comment by samu** [comment]
I've added documentation for this to the readme and will leave it at that for now.
